### PR TITLE
Separate TestInfo to another module

### PR DESCRIPTION
**Pull request by sebastienvas**
Implemented status file creation
Implemented log upload to cloud storage
Rename SetUp to Setup and TearDown to Teardown

### CODE REVIEW

File: tests/e2e/framework/framework_test.go [hunk 1]
```diff
@@ -33,7 +33,17 @@ type testConfig struct {
 	t *test
 }
 
-func NewTestConfig() *testConfig {
+func newCommonConfig(testId string) *CommonConfig {
+	t, _ := NewTestInfo(testId)
```
Comment (andraxylia) [inline tests/e2e/framework/framework_test.go]: Check the error, do not ignore it.

File: tests/e2e/framework/testInfo.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+		b, err := ioutil.ReadFile(p)
+		_, err = w.Write(b)
+		return err
+	}
+
+	walkFn := func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+		if err == nil {
```
Comment (andraxylia) [inline tests/e2e/framework/testInfo.go]: Can err be different than nil here?

File: tests/e2e/framework/testInfo.go [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+			err = uploadFile(path)
+		}
+		if err != nil {
+			glog.Warningf("An error occured when upload %s %s", path, err)
+		}
```
Comment (andraxylia) [inline tests/e2e/framework/testInfo.go]: Shouldn't you return err here?

File: tests/e2e/framework/testInfo.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+		glog.Infof("Uploading %s to gs://%s/%s", p, rp, t.Bucket)
+		o := bkt.Object(rp)
+		w := o.NewWriter(ctx)
+		defer w.Close()
+		b, err := ioutil.ReadFile(p)
+		_, err = w.Write(b)
+		return err
+	}
+
+	walkFn := func(path string, info os.FileInfo, err error) error {
```
Comment (andraxylia) [inline tests/e2e/framework/testInfo.go]: Can you clarify the desired functionality here? I am wondering if error handling should be done outside of the walk function. It's unclear why error is an input parameter.

Comment (sebastienvas) [inline tests/e2e/framework/testInfo.go]: That s the definition of the walkFn as definied here:
https://golang.org/pkg/path/filepath/#WalkFunc

File: tests/e2e/framework/testInfo.go (hunk 2 above)
Comment (sebastienvas) [inline tests/e2e/framework/testInfo.go]: I added a comment. I am filtering error as I want to go over all file in directories.

File: tests/e2e/framework/testInfo.go (hunk 3 above)
Comment (sebastienvas) [inline tests/e2e/framework/testInfo.go]: no filtering errors

File: tests/e2e/framework/framework_test.go (hunk 1 above)
Comment (sebastienvas) [inline tests/e2e/framework/framework_test.go]: Done

File: tests/e2e/framework/testInfo.go (hunk 4 above)
Comment (andraxylia) [inline tests/e2e/framework/testInfo.go]: Then it is not optimal to keep invoking walkFn is err is not nil. We can log an error and return SkipDir if the input error is not nil.

walkFn := func(path string, info os.FileInfo, err error) error {
		// We are filtering all errors here as we want filepath.Walk to go over all the files.
		if info.IsDir() {
			return nil
		}
                if err != nill {
                    glog.Warning(...)
                    return SkipDir
                }

		err = uploadFile(path)
		if err != nil {
			glog.Warningf("An error occured when upload %s %s", path, err)
		}
		return nil
	}

**Pull request merged**